runtime/src/iree/hal/command_buffer_validation.c

```diff
@@ -25,6 +25,10 @@ static iree_status_t iree_hal_command_buffer_validate_categories(
     const iree_hal_command_buffer_t* command_buffer,
     iree_hal_command_buffer_validation_state_t* validation_state,
     iree_hal_command_category_t required_categories) {
+  if (IREE_UNLIKELY(!validation_state->is_recording)) {
```

Comment:
I assume this is something you hit while debugging but is otherwise unrelated to this patch? (fine to include, just want to make sure I'm not missing something)

Reply:
yeah, this was what made splats not work - I never began the command buffer, so it never recorded the fills, and thus never changed memory 🤦 
all this validation and the one thing I didn't check was whether the command buffer was actually open for recording lol